Tests/AppcuesKitTests/PublicAPITests.swift

```diff
@@ -0,0 +1,247 @@
+//
+//  PublicAPITests.swift
+//  AppcuesKitTests
+//
+//  Created by Matt on 2023-04-05.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import XCTest
+import SwiftUI
+import AppcuesKit
+
+/// This test class exists to verify the full breadth of the public API contract. It does not assert any specific results.
+/// If this test class fails to compile, that indicates a breaking change in the public API.
+class PublicAPITests: XCTestCase {
+
+    func testAPI() throws {
+        let config = Appcues.Config(accountID: "12345", applicationID: "abc")
+            .logging(true)
+            .apiHost(URL(string: "https://api.appcues.net")!)
+            .sessionTimeout(3600)
+            .activityStorageMaxSize(25)
+            .activityStorageMaxAge(3600)
+            .urlSession(URLSession.shared)
+            .anonymousIDFactory({ UUID().uuidString })
+            .additionalAutoProperties(["test": "value"])
+            .enableUniversalLinks(true)
+
+        let appcuesInstance = Appcues(config: config)
```

Comment:
using *real* `Appcues` (and underlying `NetworkClient`) and the correct endpoint with a valid account `12345` means that these calls would actually send production data into that account, correct? Most of our tests use the `MockAppcues`and don't do any real networking - would that suffice here?

Reply:
Good point. I like the `localhost` idea, so I'll do that. `MockAppcues` would work, but it adds a layer of obfuscation on what we're trying to test here, so I'd prefer to not use that.